### PR TITLE
Allow space admins to edit settings and show gear icon for space owner

### LINKED ISSUE #287: Can't see gear icon in space
**Describe the bug**
The gear icon to get to space settings isn't showing for some users.

**Expected behavior**
When going to a space that you own, you should see a gear icon in the top right.

### PR DESCRIPTION
**Pull request by samuveth**
Fixes #287

### CODE REVIEW

File: src/views/Proposals.vue [hunk 1]
```diff
@@ -21,7 +21,7 @@
           <UiButton>{{ $t('proposals.new') }}</UiButton>
         </router-link>
         <router-link
-          v-if="isMember && isEns"
+          v-if="(isAdmin && isEns) || (isOwner && isEns)"
```
Comment (ChaituVR) [inline src/views/Proposals.vue:24]: all admins will have ens names?

Comment (bonustrack) [reply src/views/Proposals.vue:24]: No, but here the "isEns" is about the space, not about the admin

Comment (samuveth) [reply src/views/Proposals.vue:24]: I think this is checking the domain of the space, not the if admin has ENS.

Comment (samuveth) [reply src/views/Proposals.vue:24]:
```
 key() {
      return this.domain || this.$route.params.key;
    },```

File: src/views/Proposals.vue [hunk 2]
```diff
@@ -115,12 +117,15 @@ export default {
           .sort((a, b) => b[1].msg.payload.end - a[1].msg.payload.end, 0)
       );
     },
-    isMember() {
-      const members = this.space.members.map(address => address.toLowerCase());
+    isOwner() {
+      return this.spaceUri.includes(this.web3.account);
+    },
+    isAdmin() {
+      const admins = this.space.admins.map(address => address.toLowerCase());
```
Comment (ChaituVR) [inline src/views/Proposals.vue:124]: @samuveth Getting error if no admins field in space setting
image

Comment (samuveth) [reply src/views/Proposals.vue:124]: Hm, I wonder why it works for `isMember` like that.

Comment (samuveth) [reply src/views/Proposals.vue]: If `members` is always defined then `admins` should be too no?

Comment (bonustrack) [reply src/views/Proposals.vue]: If admins is always defined, then 99% of the space will fail to load, cuz the hub check if all space settings are correct. Both should be option fields.

Comment (samuveth) [reply src/views/Proposals.vue]: @bonustrack I mean like members, it's always at minimum `members: []` when the space setting are loaded from the hub. Or is that not the case?

Comment (bonustrack) [reply src/views/Proposals.vue]: It's not necessary the case

Comment (samuveth) [reply src/views/Proposals.vue]: Then some spaces must be getting an error here: https://github.com/snapshot-labs/snapshot/blob/7b6d7c80d408596dda5f0192e4286b9a0018246d/src/views/Proposals.vue#L115

Comment (bonustrack) [reply src/views/Proposals.vue]: Not with that https://github.com/snapshot-labs/snapshot/blob/a53474ba13edd8ffbf480c71ace3e2c7b8da885f/src/helpers/utils.ts#L87

Comment (samuveth) [reply src/views/Proposals.vue]: That's what I was looking for!

**Comment by samuveth** [comment]
[UI: allow "admins" to edit space settings (except for the field "admins")](https://trello.com/c/9tX5Lvzx/50-ui-allow-admins-to-edit-space-settings-except-for-the-field-admins)

File: src/views/Proposals.vue [hunk 3]
```diff
@@ -129,6 +135,7 @@ export default {
     this.tab =
       this.$route.params.tab || this.space.filters.defaultTab || this.tab;
     this.proposals = await getProposals(this.space);
+    this.spaceUri = await getSpaceUri(this.key);
```
Comment (bonustrack) [inline src/views/Proposals.vue:138]: This call is a bit heavy (require 3 requests and also loading of ENS pkg), i think it's better to not rely on it and only show settings button for admins.

**Review by bonustrack — APPROVED** [review]
Looks good!

**Pull request merged**